linkerd/meshtls/rustls/src/client.rs

```diff
@@ -72,15 +72,23 @@ impl Connect {
             .expect("identity must be a valid DNS name");
 
         Self {
+            server_id: client_tls.server_id.into(),
             server_name,
             config,
         }
     }
 }
 
+fn extract_cert(c: &rustls::ClientConnection) -> io::Result<&rustls::Certificate> {
```

Comment:
i'm not sure if `io::Error` is really the right return type for this, since the error is not actually IO-related...should we maybe instead change the `Service::Error` associated type for `Connect`? Or, does that break some other thing?